### DISCUSSION TITLE
Start android app with arguments?

### DISCUSSION THREAD

**Discussion by ngocdh** [jamulussoftware/jamulus#1382] [Technical Support & Help]
Hi,
I could start Jamulus from termux, like this

```bash
am start -n com.github.jamulussoftware.jamulus/org.qtproject.qt5.android.bindings.QtActivity
```

However, I couldn't find a way to add arguments, for example "--connect SERVERIP"

My intent was write a script to detect a jamulus server in local network, and open Jamulus using "--connect" flag to connect to the found server. Even without the detection part, this could help me avoid typing the server address every time, because the app now doesn't memorize settings.

Anyone knows how to add arguments to open an android app? Or maybe this is not possible?

**Comment by ann0see** [comment]
Not sure if that's possible. I doubt we have anybody with that deep Android knowledge. Maybe try to ask on the Qt forum or something like XDA Developers?

**Reply by ngocdh** [reply]
Thanks @ann0see. I tried searching the Qt forum, nothing so far. Will update here if I find anything.

**Comment by DominikSchaller** [comment]
Weird, in my Android app version the settings are stored.

**Reply by ngocdh** [reply]
Oh? In my app, nothing is stored: name, country,  audio setup, network setup, server address, etc. 
Google pixel 5. I tried reinstall but nothing changed.

**Comment by ngocdh** [comment]
Update: seems like we need to use "android.app.arguments" in AndroidManifest.xml
Ref: https://wiki.qt.io/AndroidServices

**Comment by ann0see** [comment]
@ngocdh I think I could reproduce the "Jamulus doesn't save settings on Android" bug: If I swype and close the app after having made changes, to me it seems as if nothing gets saved. But if I go to File>Close after re-opening the app, settings are saved. I think the closing process is more like a kill then a normal close signal on Android --> Changes aren't saved. That's similar to killing the Program via Task-Manager or ctrl-c.
I think this is a bug.

**Reply by pljones** [reply]
Separate issue?

**Reply by ann0see** [reply]
Yes.
